### DISCUSSION TITLE
Newbie control of an Existing Rez Install

### DISCUSSION THREAD

**Discussion by Saatchi3D** [AcademySoftwareFoundation/rez#1069] [Q&A] [answered]
I apologize for this newbie question(s).  We need to migrate from Python 2 to Python 3 and Rez to current.  First some details:

Air-gapped, Windows 10, VFX network
Rez: 2.16.  Installed 2017(?)
Python 2.7

The programmer that installed and initially configured Rez is gone.  I'm the team lead and am starting the migration process.  My goal is to create a "stand alone" env with Python 3, and Qt/PySide for my team to use to do the migration and testing to Python 3.

I've installed Rez 2.83 and python 3.9 and, for the most part, the beginnings of a standalone env.  I say "mostly", because references to rez 2.16 still show up in the pathing created by 'rez env' so I'm guessing I don't have rez 2.83 installed correctly.

**My first question**, of many, what is the proper upgrade procedure?  I created a .cmd file that starts a new command session with the rez environment variables pointing to a local copy 2.83 and PATH set appropriately.  This is so I can setup the test environment without disrupting my day to day work.  I'm taking guesses and black boxing this whole migration process.  If you can point me to doc, examples, give suggestions, that would be most helpful.

I am a big fan of Rez and am hoping that upgrading will improve some of the inherent performance issues with the set up environments (PATH and PYTHONPATH length).  I've been reading some of the discussion on that on this forum.  What ever I learn I can document and pass on to the rest of my team so the knowledge isn't lost again.

Thanks
John Morrison
3D Programmer - Lead
ThePub
Publicis Groupe

**Comment by davidlatwe** [comment]
> I've installed Rez 2.83 and python 3.9 and, for the most part, the beginnings of a standalone env. I say "mostly", because references to rez 2.16 still show up in the pathing created by 'rez env'

Hi @Saatchi3D , could you provide a bit more detail on this ? You mean the the `rez --version` print out different version once inside a `rez-env` resolved context ?

**Comment by Saatchi3D** [comment, marked as answer]
Hi David.

Sorry to be such a bad newbie.  I found the problem and corrected it.  I didn't update the forum entry.

The old Rez was showing up in the PATH ahead of the new one.   Here's why....

I created script (setupPy3.cmd) to START a new command session with the environment vars set for Rez 2.83, and the PATH pointing to Rez 2.83 and Python 3.9.  Rez -V showed 2.83.0 as it should.  Python -V showed 3.9 also as it should.  This all worked fine until I did a rez env.  At this point, since Rez STARTs a new command session,  the old PATH which included Rez 2.16 got pulled back in.

The correction was to remove Rez 2.16 from the PATH at the top (Windows) level so any session started would not get it.  I then created an additional "setupPy2" script for Python 2 which included Rez 2.16.  This solved the problem for me.

We are working on the monumental task of converting from Python 2 to Python 3.  I'm creating a test environment so the rest of my team can migrate our code base and apps.

I hope the long-winded explanation above makes sense.

I'll be back here again, I sure, as I run into issues due to lack of knowledge.

Thank you.
John Morrison
________________________________

**Reply by instinct-vfx** [reply]
For some context, the issue here is likely that Rez is actually pulling PATH from the registry rather than the parent shell. The reason is to make sure the Rez execution environment, the resolved environments and nested environments are strictly separated. If this was not the case you might end up in situations that have conflicting variants on PYTHONPATH for example.

**Comment by Saatchi3D** [comment]
Thanks Thorsten,

If I had thought about it, I should have realized that Windows keeps the "official" PATH(s) in the registry.  And I should have said that that was the PATH that I removed the Rez 2.16 pathing from.

Thanks
John Morrison
________________________________
